### PR TITLE
Improved build of tau and related dependencies

### PR DESCRIPTION
**Pull request by alalazo**
Modifications :
- [X] added support for external packages to tau build
- [X] updated scorep build (based on unwrapped compiler, rather than unsupported files)
- [X] added pdt
- [X] updated versions of opari2, otf2, cube

Manually tested:
- Ubuntu 14.04 and RHLE 6 with GCC and MPICH

### CODE REVIEW

**Comment by alalazo** [comment]
Please double check as the build process required a lot of tweaking.

**Comment by tgamblin** [comment]
@khuck: How does this package look to you?  I think you know more about deploying TAU than I do :)

**Comment by khuck** [comment]
I think it looks mostly fine - although I would make ScoreP a variant, not a dependency.

**Comment by tgamblin** [comment]
@khuck @alalazo : ok I can merge once that's added.  Thanks!

**Comment by alalazo** [comment]
@tgamblin @khuck : done.

File: var/spack/packages/scorep/package.py [hunk 1]
```diff
@@ -1,74 +1,80 @@
...
-    mpi_user_provided = """\
-MPICC=mpicc
-MPICXX=mpicxx
-MPIF77=mpif77
-MPIFC=mpif90
-MPI_CFLAGS=-fPIC
-MPI_CXXFLAGS=-fPIC
-"""
+    def get_compiler_config_line(self):
+        backend_user_provided = ['CC=%s' % self.compiler.cc_names[0],
```
Comment (alalazo) [inline var/spack/packages/scorep/package.py]: I guess this and other similar pieces of code can be simplified or even removed as soon as #255 will be merged

Comment (tgamblin) [inline var/spack/packages/scorep/package.py]: @alalazo: I think that's true.
@daboehme: can you look at this?  Will ScoreP detect things automatically now that compilers will have appropriate executable names?  See #255.

**Comment by tgamblin** [comment]
Looks good to me!

File: var/spack/packages/scorep/package.py (hunk 1 above)
Comment (daboehme) [inline var/spack/packages/scorep/package.py]: So #255 should make the bizarre "guess the c++ compiler from the c compiler name" hack in Score-P's build system  no longer break, but you will still have to specify the compiler explicitly. You could do that now with --with-nocross-compiler-suite=[gnu|intel|whatever], but the way here is probably easier and more generic.

**Pull request merged**